### PR TITLE
btrfs: Check that the current kernel version will not have issues down the line

### LINKED ISSUE #47: Btrfs bug tracking
An issue to log Btrfs deadlock stack-traces:
```
[1036019.465941] INFO: task ocluster-worker:3435002 blocked for more than 120 seconds.
[1036019.473678]       Not tainted 5.4.0-54-generic #60-Ubuntu
[1036019.480023] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[1036019.488751] ocluster-worker D    0 3435002      1 0x00000004
[1036019.488758] Call Trace:
[1036019.488772]  __schedule+0x2e3/0x740
[1036019.488776]  schedule+0x42/0xb0
[1036019.488780]  rwsem_down_read_slowpath+0x16c/0x4a0
[1036019.488786]  down_read+0x85/0xa0
[1036019.488791]  lookup_slow+0x2d/0x60
[1036019.488795]  walk_component+0x1da/0x360
[1036019.488798]  ? link_path_walk.part.0+0x6d/0x550
[1036019.488803]  path_lookupat.isra.0+0x80/0x230
[1036019.488806]  ? __switch_to_asm+0x40/0x70
[1036019.488826]  ? __switch_to_asm+0x34/0x70
[1036019.488836]  ? __switch_to_asm+0x40/0x70
[1036019.488847]  ? __switch_to_asm+0x34/0x70
[1036019.488856]  ? __switch_to_asm+0x40/0x70
[1036019.488867]  filename_lookup+0xae/0x170
[1036019.488880]  ? __check_object_size+0x13f/0x150
[1036019.488892]  ? strncpy_from_user+0x4c/0x150
[1036019.488902]  user_path_at_empty+0x3a/0x50
[1036019.488906]  vfs_statx+0x7d/0xe0
[1036019.488918]  __do_sys_newlstat+0x3e/0x80
[1036019.488933]  ? do_user_addr_fault+0x216/0x450
[1036019.488941]  ? __do_page_fault+0x58/0x90
[1036019.488944]  __x64_sys_newlstat+0x16/0x20
[1036019.488950]  do_syscall_64+0x57/0x190
[1036019.488954]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[1036019.488959] RIP: 0033:0x7f9d5fc6c6ea
[1036019.488968] Code: Bad RIP value.
[1036019.488970] RSP: 002b:00007ffebcce6ec8 EFLAGS: 00000246 ORIG_RAX: 0000000000000006
[1036019.488973] RAX: ffffffffffffffda RBX: 00005599c1807060 RCX: 00007f9d5fc6c6ea
[1036019.488975] RDX: 00007ffebcce6f20 RSI: 00007ffebcce6f20 RDI: 00005599c1807060
[1036019.488977] RBP: 0000000000000000 R08: 0000000000000001 R09: 00007f9d5f5c8f10
[1036019.488978] R10: 00005599bec07d60 R11: 0000000000000246 R12: 00005599bffac4e0
[1036019.488980] R13: 00007ffebcce6f20 R14: 00005599c17302a0 R15: 00007f9d5f5c8828
```

### PR DESCRIPTION
**Pull request by kit-ty-kate**
Fixes #47

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
This check also seems to "fix" (or at least make less common) the issue opam-repo-ci was experiencing, described in https://github.com/ocurrent/ocurrent/pull/261#issuecomment-822361754

File: lib/btrfs_store.ml [hunk 1]
```diff
@@ -87,7 +87,27 @@ let purge path =
...
+      | Some maj, Some min when (maj, min) >= (5, 8) ->
+          Lwt.return_unit
+      | Some maj, Some min ->
+          Lwt.fail_with
+            (Fmt.str
+               "You need at least linux 5.8 to use the btrfs backend, \
+                but current kernel version is '%d.%d'"
+               maj min)
+      | _, _ ->
+          Lwt.fail_with "Could not parse kernel version"
```
Comment (talex5) [inline lib/btrfs_store.ml:104]:
```suggestion
          Fmt.failwith "Could not parse kernel version %S" kver
```

File: lib/btrfs_store.ml [hunk 2]
```diff
@@ -87,7 +87,27 @@ let purge path =
...
+      end
+  | _ ->
+      Lwt.fail_with "Could not parse output of 'uname -r'"
```
Comment (talex5) [inline lib/btrfs_store.ml:107]:
```suggestion
      Fmt.failwith "Could not parse output of 'uname -r' (%S)" kver
```

**Review by talex5 — COMMENTED** [review]
Seems reasonable. Btrfs only works on Linux, so this should be fine for now.

**Comment by kit-ty-kate** [comment]
I've added a commit that reverts https://github.com/ocurrent/obuilder/pull/43 because I believe this should not be required anymore and might help some high-core-count workers experiencing slowdowns (e.g. `pima`)

**Comment by talex5** [comment]
Looks good - the Travis tests need updating to test on a newer Ubuntu, though!

**Comment by kit-ty-kate** [comment]
Sadly Travis-CI doesn't seem to have support for non-LTS version of Ubuntu. Github actions doesn't seem to support that either. What should we do?

**Comment by talex5** [comment]
Oh. Maybe add a fake `uname` script to `$PATH` during the tests?

**Comment by kit-ty-kate** [comment]
> Oh. Maybe add a fake `uname` script to `$PATH` during the tests?

Thanks for the idea! This is done and CI now passes.

**Comment by talex5** [comment]
Thanks!

**Pull request merged**